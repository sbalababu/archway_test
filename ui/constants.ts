export const TOKEN_EXTRACTOR = (s: any) => s.getIn(['login', 'token']);

export const RECENT_WORKSPACES_KEY = 'recentWorkspaces';
export const SPNEGO = 'spnego';
export const workspaceStatuses = ['approved', 'pending', 'rejected'];
export const workspaceBehaviors = ['simple', 'structured', 'custom'];
export const behaviorProperties = {
  simple: {
    title: 'Simple',
    icon: 'team',
  },
  structured: {
    title: 'Structured',
    icon: 'deployment-unit',
  },
  custom: {
    title: 'Custom',
    icon: 'select',
  },
};

export enum FeatureFlagType {
  Application = 'application',
  Messaging = 'messaging',
  DiskQuota = 'disk-quota',
}

export enum ProvisioningType {
  Pending = 'pending',
  Complete = 'complete',
}

export enum ModalType {
  DeprovisionWorkspace = 'deprovisionWorkspace',
  ProvisionWorkspace = 'provisionWorkspace',
  DeleteWorkspace = 'deleteWorkspace',
  Application = 'application',
  Kafka = 'kafka',
  SimpleTopicMember = 'simpleTopicMember',
  SimpleMember = 'simpleMember',
  ChangeOwner = 'changeOwner',
  ModifyDiskQuota = 'modifyDiskQuota',
  ModifyCoreMemory = 'modifyCoreMemory',
}

export enum NotificationType {
  Success = 'success',
  Info = 'info',
  Warning = 'warning',
  Error = 'error',
}

export enum ComplianceType {
  PCI = 'pci',
  PII = 'pii',
  PHI = 'phi',
}

export enum LinksGroupCardPage {
  Overview = 'home',
  Manage = 'manage',
}
