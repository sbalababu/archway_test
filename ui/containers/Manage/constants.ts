export enum ManagePage {
  ComplianceTab = 'compliances',
  LinksTab = 'custom-links',
}

export enum OpenType {
  Add = 'add',
  Update = 'update',
  Delete = 'delete',
}

export const formLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 4 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 20 },
    lg: { span: 16 },
  },
};

export const linkLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 3 },
  },
  wrapperCol: {
    xs: { span: 24 },
    md: { span: 21 },
    lg: { span: 15 },
  },
};
