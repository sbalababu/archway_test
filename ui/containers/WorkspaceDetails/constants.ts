export enum ShowTypes {
  Users = 'Users',
  Groups = 'Groups',
}
