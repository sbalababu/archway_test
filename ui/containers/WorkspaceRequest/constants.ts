export const PAGE_BEHAVIOR = 'behavior';
export const PAGE_CUSTOM_WORKSPACES = 'custom workspaces';
export const PAGE_DETAILS = 'details';
export const PAGE_COMPLIANCE = 'compliance';
export const PAGE_REVIEW = 'review';
