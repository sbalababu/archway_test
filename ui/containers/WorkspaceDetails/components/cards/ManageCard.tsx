import * as React from 'react';
import { Card, Button } from 'antd';
import { ModalType } from '../../../../constants';

interface Props {
  children?: any;
  title: string;
  buttonText: string;
  modalType: ModalType;
  disabled: boolean;
  showModal: (e: React.MouseEvent, type: ModalType) => void;
}

const WorkspaceCard = ({ title, children, modalType, buttonText, disabled, showModal }: Props) => {
  return (
    <Card style={{ height: '100%' }} bordered>
      <div style={{ textAlign: 'center', fontSize: 17 }}>{title}</div>
      <div style={{ textAlign: 'left', fontSize: 15, padding: '20px 20px 100px 20px' }}>{children}</div>
      <div
        style={{
          fontWeight: 300,
          position: 'absolute',
          bottom: 20,
          left: '50%',
          transform: 'translateX(-50%)',
          padding: 20,
        }}
      >
        <Button type="primary" onClick={e => showModal(e, modalType)} disabled={disabled}>
          {buttonText}
        </Button>
      </div>
    </Card>
  );
};

export default WorkspaceCard;
