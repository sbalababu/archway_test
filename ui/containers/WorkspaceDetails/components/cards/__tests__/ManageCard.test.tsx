import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ManageCard from '../ManageCard';
import { ModalType } from '../../../../../constants';

describe('ManageCard', () => {
  it('it renders correctly', () => {
    const props = {
      title: 'Provision',
      buttonText: 'Provision',
      modalType: ModalType.ProvisionWorkspace,
      disabled: false,
      showModal: () => null,
    };
    const wrapper = shallow(<ManageCard {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
