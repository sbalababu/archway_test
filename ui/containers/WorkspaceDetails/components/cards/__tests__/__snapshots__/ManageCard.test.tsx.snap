// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManageCard it renders correctly 1`] = `
<Card
  bordered={true}
  style={
    Object {
      "height": "100%",
    }
  }
>
  <div
    style={
      Object {
        "fontSize": 17,
        "textAlign": "center",
      }
    }
  >
    Provision
  </div>
  <div
    style={
      Object {
        "fontSize": 15,
        "padding": "20px 20px 100px 20px",
        "textAlign": "left",
      }
    }
  />
  <div
    style={
      Object {
        "bottom": 20,
        "fontWeight": 300,
        "left": "50%",
        "padding": 20,
        "position": "absolute",
        "transform": "translateX(-50%)",
      }
    }
  >
    <Button
      block={false}
      disabled={false}
      ghost={false}
      htmlType="button"
      loading={false}
      onClick={[Function]}
      type="primary"
    >
      Provision
    </Button>
  </div>
</Card>
`;
